api/v1beta1/network_disruption.go

```diff
@@ -150,23 +172,118 @@ type NetworkDisruptionCloudServiceSpec struct {
 	ConnState string `json:"connState,omitempty"`
 }
 
-// Validate validates args for the given http filters.
-func (s *NetworkHTTPFilters) Validate() error {
-	if s.Path != "" {
-		if len(s.Path) > MaxNetworkPathCharacters {
-			return fmt.Errorf("the path specification at the network disruption level is not valid; should not exceed 100 characters")
+func (p HTTPPath) validate() error {
+	if len(p) > MaxNetworkPathCharacters {
+		return fmt.Errorf("the paths specification at the network disruption level is not valid; should not exceed 100 characters")
+	}
+
+	if regexp.MustCompile(`\s`).MatchString(string(p)) {
+		return fmt.Errorf("the paths specification at the network disruption level is not valid; should not contains spaces")
+	}
+
+	if string(p[0]) != DefaultHTTPPathFilter {
+		return fmt.Errorf("the paths specification at the network disruption level is not valid; should start with a /")
+	}
+
+	return nil
+}
+
+func (paths HTTPPaths) isNotDefault() bool {
+	if len(paths) == 0 || len(paths) > 1 {
+		return false
+	}
+
+	return paths[0] != DefaultHTTPPathFilter
+}
+
+func (h HTTPMethods) isNotEmpty() bool {
+	return len(h) >= 1
+}
+
+// validate validates args for the given http filters.
+func (s *NetworkHTTPFilters) validate() (retErr error) {
+	if s.DeprecatedPath != "" {
+		retErr = multierror.Append(retErr, fmt.Errorf("the Path specification at the HTTP network disruption level is deprecated; use Paths HTTP field instead"))
+	}
+
+	if s.DeprecatedMethod != "" {
+		retErr = multierror.Append(retErr, fmt.Errorf("the Method specification at the HTTP network disruption level is deprecated; use Methods HTTP field instead"))
+	}
+
+	if len(s.Paths) > MaxNetworkPaths {
+		retErr = multierror.Append(retErr, fmt.Errorf(HTTPPathsFilterErrorPrefix+"the number of paths must not be greater than %d; Number of paths: %d", MaxNetworkPaths, len(s.Paths)))
+	} else if len(s.Paths) > 0 {
+		visitedPaths := make(map[HTTPPath]struct {
```

Comment:
Could we break this entire block under `else if len(s.Paths) > 0` out into a function w/ unit tests?

Reply:
A good practice is to unit tests only public functions. Private functions are tested implicitely and usually come from the refactoring step. In this situation the unit tests are dedicated to the public function `Validate` of the `NetworkDisruptionSpec`.  Source https://sd.blackball.lv/library/unit_testing_(2020).pdf page 260 `11.1 Unit testing private methods`